### PR TITLE
Update to Protocol 17.0.0 (Fixes #13125)

### LINKED ISSUE #13125: Update to Protocol 17.0.0
## Description

https://github.com/mozilla/protocol/releases/tag/v17.0.0

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

- Updates `@mozilla-protocol/core` to v17.0.0.
- Removes `window.Mzp.*` global namespace.
- Imports Protocol JS components via `import/require`.

## Significant changes and points to review

This is a fairly large change that effects quite a number of pages, so could use some careful review.

## Issue / Bugzilla link

#13125

## Testing

Demo: https://www-demo3.allizom.org/en-US/

Successful test run: https://github.com/mozilla/bedrock/actions/runs/4958556081

Modals open as expected:

- [x] http://localhost:8000/en-US/
- [x] http://localhost:8000/en-US/careers/
- [x] http://localhost:8000/en-US/about/leadership/
- [x] http://localhost:8000/en-US/diversity/2021/what-we-build/
- [x] http://localhost:8000/en-US/firefox/all/
- [x] http://localhost:8000/en-US/firefox/features/tips/
- [x] http://localhost:8000/en-US/firefox/new/ (requires Firefox browser when clicking download and signed out of FxA)
- [x] http://localhost:8000/en-US/firefox/welcome/6/
- [x] http://localhost:8000/en-US/foundation/annualreport/2021/

Side menu toggles as expected (small viewports)

- [x] http://localhost:8000/en-US/about/webvision/full/
- [x] http://localhost:8000/en-US/contact/spaces/
- [x] http://localhost:8000/en-US/privacy/

Navigation functions as expected:

- [x] http://localhost:8000/en-US/

Footer details toggle as expected (small viewports)

- [x] http://localhost:8000/en-US/

Language switcher works as expected:

- [x] http://localhost:8000/en-US/

Sub navigation toggles as expected (small viewports)

- [x] http://localhost:8000/en-US/products/vpn/
- [x] http://localhost:8000/en-US/firefox/family/

Download dropdown menus toggle as expected:

- [x] http://localhost:8000/en-US/firefox/
- [x] http://localhost:8000/en-US/firefox/browsers/

Sticky promo appears on scroll as expected:

- [x] http://localhost:8000/en-US/firefox/new/

Details menus toggle as expected (small viewports)

- [x] http://localhost:8000/en-US/about/policy/transparency/jul-dec-2022/

Newsletter form expands on focus:

- [x] http://localhost:8000/en-US/

**Did I miss anything else?**

### CODE REVIEW

File: media/js/base/protocol-globals.es6.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import MzpSupports from '@mozilla-protocol/core/protocol/js/supports';
+import MzpUtils from '@mozilla-protocol/core/protocol/js/utils';
+import MzpDetails from '@mozilla-protocol/core/protocol/js/details';
+
+window.MzpSupports = MzpSupports;
+window.MzpUtils = MzpUtils;
+window.MzpDetails = MzpDetails;
```
Comment (nathan-barrett) [inline media/js/base/protocol-globals.es6.js:13]: **Curious: ** Why are these three the only ones being added to the window here? Noticed that others like the `MzpLangSwitcher` aren't

File: webpack.config.js [hunk 2]
```diff
@@ -17,9 +17,6 @@ const webpack = require('webpack');
 
 function resolveBundles(fileList) {
     return fileList.map((f) => {
-        if (f.match(/^protocol\//)) {
-            return `./node_modules/@mozilla-protocol/core/${f}`;
-        }
```
Comment (nathan-barrett) [inline webpack.config.js:22]: nice to see this removed!

**Review by nathan-barrett — APPROVED** [review]
R+ - awesome work Alex, everything ran smoothly for me and I didn't run into any issues while checking locally or on the demo. Happy to see this happening

File: media/js/base/protocol-globals.es6.js (hunk 1 above)
Comment (alexgibson) [reply media/js/base/protocol-globals.es6.js:13]: These are all modules that are commonly used by many scripts throughout the site, so it makes sense to bundle them once in a shared script. If we imported them every time a file needed them, then we'd likely end up with larger bundles and duplicated code. Does that make sense?

**Pull request merged**